extensions/quartz/deployment/src/main/java/io/quarkus/quartz/deployment/QuartzProcessor.java

```diff
@@ -118,19 +126,51 @@ QuartzJDBCDriverDialectBuildItem driver(List<JdbcDataSourceBuildItem> jdbcDataSo
                     "The Agroal extension is missing and it is required when a Quartz JDBC store is used.");
         }
 
-        Optional<JdbcDataSourceBuildItem> selectedJdbcDataSourceBuildItem = jdbcDataSourceBuildItems.stream()
-                .filter(i -> config.dataSourceName.isPresent() ? config.dataSourceName.get().equals(i.getName())
-                        : i.isDefault())
-                .findFirst();
-
-        if (!selectedJdbcDataSourceBuildItem.isPresent()) {
-            String message = String.format(
-                    "JDBC Store configured but the '%s' datasource is not configured properly. You can configure your datasource by following the guide available at: https://quarkus.io/guides/datasource",
-                    config.dataSourceName.isPresent() ? config.dataSourceName.get() : "default");
-            throw new ConfigurationException(message);
+        Optional<String> driverDelegate = config.driverDelegate;
+        if (driverDelegate.isPresent()) {
+            // user-specified custom delegate
+            ClassInfo customDelegate = indexBuildItem.getIndex().getClassByName(driverDelegate.get());
+            if (customDelegate == null) {
+                String message = String.format(
+                        "Custom JDBC delegate implementation class '%s' was not found in Jandex index. " +
+                                "Make sure the dependency containing this class has proper marker file enabling discovery. " +
+                                "Alternatively, you can index a dependency using IndexDependencyBuildItem.",
+                        driverDelegate.get());
+                throw new ConfigurationException(message);
+            } else {
+                // any custom implementation needs to be a subclass of known Quarkus delegate
+                Set<DotName> delegateDotNames = Set.of(DELEGATE_MSSQL, DELEGATE_POSTGRESQL, DELEGATE_DB2V8, DELEGATE_STDJDBC,
+                        DELEGATE_HSQLDB);
+                Type superClassType = customDelegate.superClassType();
+                boolean implementsKnownDelegate = false;
+                while (!superClassType.name().equals(DotName.OBJECT_NAME)) {
+                    if (delegateDotNames.contains(superClassType.name())) {
```

Comment:
This means that the custom delegate class must be a direct subclass of a Quarkus type. I think that it's ok but maybe also too limiting?

CC @arpitmotwani15

Reply:
Wait, that's an oversight, it should be any subclass of a known implementation.
Let me fix that.

Reply:
@mkouba should be fixed now